src/java/org/apache/cassandra/cql3/restrictions/MergedRestriction.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.restrictions;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.RangeSet;
+
+import org.apache.cassandra.cql3.Operator;
+import org.apache.cassandra.cql3.QueryOptions;
+import org.apache.cassandra.cql3.functions.Function;
+import org.apache.cassandra.db.filter.RowFilter;
+import org.apache.cassandra.index.Index;
+import org.apache.cassandra.index.IndexRegistry;
+import org.apache.cassandra.schema.ColumnMetadata;
+
+import static org.apache.cassandra.cql3.statements.RequestValidations.invalidRequest;
+
+/**
+ * A {@code SingleRestriction} which is the result of merging multiple {@code SimpleRestriction} applying to
```

Comment:
```suggestion
 * A {@link SingleRestriction} which is the result of merging multiple {@ link SimpleRestriction} applying to
```
nit: This makes IDE navigation a little nicer, FWIW

Reply:
There are lots more of these around the patch, but I'll let you do what you like w/ them.

Reply:
I see what you mean. It seems that the normal use of link is to create an hyperlink in your javadoc on the first occurrence of a class or method name in the doc. I will not go back through all the doc at this point but I will do it in the future.

Reply:
Yeah, no worries. It can be a nice navigation aid.